src/vsphere_cpi/lib/cloud/vsphere/resources/vm.rb

```diff
@@ -389,22 +389,27 @@ def move_disks(disks_to_move)
             dest_filename = original_disk_path.match(/^\[[^\]]+\] (.*)/)[1]
           else
             # Use current disk path incuded with the disk
-            dest_filename =  disk.backing.file_name.match(/^\[[^\]]+\] .*\/(.*)/)[1]
+            dest_filename = disk.backing.file_name.match(/^\[[^\]]+\] .*\/(.*)/)[1]
             dest_filename = "#{disk_path}/#{dest_filename}"
           end
           dest_path = "[#{current_datastore}] #{dest_filename}"
 
-          @client.move_disk(datacenter_mob, disk.backing.file_name, datacenter_mob, dest_path)
+          destination_exists = @client.disk_path_exists?(@mob, dest_path)
+          datastore_type = disk.backing.datastore.summary.type
```

Comment:
Can we add a comment to explain the background for this and why we think this is safe? I think its reasonable but in 2 years im 100% sure i wont remember a word of it.